NebulaHost/PacketProcessors/GameHistory/GameHistoryResearchContributionProcessor.cs

```diff
@@ -13,7 +14,11 @@ public void ProcessPacket(GameHistoryResearchContributionPacket packet, NebulaCo
             //Check if client is contributing to the correct Tech Research
             if (packet.TechId == GameMain.history.currentTech)
             {
+                Log.Info($"ProcessPacket researchContribution: got package for same tech");
                 GameMain.history.AddTechHash(packet.Hashes);
+                PlayerManager playerManager = MultiplayerHostSession.Instance.PlayerManager;
+                playerManager.GetPlayer(conn).UpdateResearchProgress(packet.TechId, packet.Hashes);
+                Log.Info($"ProcessPacket researchContribution: playerid by: {playerManager.GetPlayer(conn).Id} - hashes {packet.Hashes}");
```

Comment:
Do we maybe want to make this a debug log instead? Could get spammy

Reply:
Good catch, I missed that one